solver/src/main/java/org/chocosolver/solver/constraints/real/RealPowerPropagator.java

```diff
@@ -0,0 +1,108 @@
+package org.chocosolver.solver.constraints.real;
+
+import org.chocosolver.solver.constraints.Propagator;
+import org.chocosolver.solver.constraints.PropagatorPriority;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.expression.continuous.arithmetic.RealIntervalConstant;
+import org.chocosolver.solver.variables.RealVar;
+import org.chocosolver.solver.variables.events.RealEventType;
+import org.chocosolver.util.ESat;
+import org.chocosolver.util.objects.RealInterval;
+import org.chocosolver.util.tools.RealUtils;
+
+/**
+ * It's a pure java alternative to ibex that must be used to calculate
+ * power operations (y = x^k) with integer positive exponents (k).
+ *
+ * <p>
+ * Just in case, this is an external library for interval arithmetic.
+ * Reference: https://jitpack.io/p/jinterval/jinterval#readme
+ * <p>
+ * This implementation is based from the chapter 16.1 "From Discrete to
+ * Continuous Constraints", example 16.2 [1].
+ * <p>
+ * [1] Francesca Rossi, Peter van Beek, and Toby Walsh. 2006. Handbook of
+ * Constraint Programming. Elsevier Science Inc., USA.
+ *
+ * @author João Pedro Schmitt
+ * @since 08/09/2020
+ */
+public class RealPowerPropagator extends Propagator<RealVar> {

```

Comment:
In Choco, we name propagator with prefix `Prop`.
You could change the name to `PropRealPower`.